src/Core/Services/GraphQLSchemaCreator.cs

```diff
@@ -251,6 +265,225 @@ private DocumentNode GenerateSqlGraphQLObjects(RuntimeEntities entities, Diction
+                }
+
+                string dataSourceName = _runtimeConfigProvider.GetConfig().GetDataSourceNameFromEntityName(sourceEntityName);
+                ISqlMetadataProvider sqlMetadataProvider = _metadataProviderFactory.GetMetadataProvider(dataSourceName);
+                SourceDefinition sourceDefinition = sqlMetadataProvider.GetSourceDefinition(sourceEntityName);
+                Dictionary<string, FieldDefinitionNode> sourceFieldDefinitions = sourceObjectTypeDefinitionNode.Fields.ToDictionary(field => field.Name.Value, field => field);
+
+                foreach((_, EntityRelationship relationship) in entity.Relationships)
+                {
+                    string targetEntityName = relationship.TargetEntity;
+                    if (!string.IsNullOrEmpty(relationship.LinkingObject))
+                    {
+                        // For M:N relationships, the fields in the entity are always referenced fields.
+                        continue;
+                    }
+
+                    if (// Retrieve all the relationship information for the source entity which is backed by this table definition
+                        sourceDefinition.SourceEntityRelationshipMap.TryGetValue(sourceEntityName, out RelationshipMetadata? relationshipInfo) &&
+                        // From the relationship information, obtain the foreign key definition for the given target entity
+                        relationshipInfo.TargetEntityToFkDefinitionMap.TryGetValue(targetEntityName, out List<ForeignKeyDefinition>? listOfForeignKeys))
+                    {
+                        sqlMetadataProvider.GetEntityNamesAndDbObjects().TryGetValue(sourceEntityName, out DatabaseObject? sourceDbo);
+                        // Find the foreignkeys in which the source entity is the referencing object.
+                        IEnumerable <ForeignKeyDefinition> sourceReferencingForeignKeysInfo =
+                            listOfForeignKeys.Where(fk =>
+                                fk.ReferencingColumns.Count > 0
+                                && fk.ReferencedColumns.Count > 0
+                                && fk.Pair.ReferencingDbTable.Equals(sourceDbo));
+
+                        sqlMetadataProvider.GetEntityNamesAndDbObjects().TryGetValue(targetEntityName, out DatabaseObject? targetDbo);
+                        // Find the foreignkeys in which the target entity is the referencing object, i.e. source entity is the referenced object.
+                        IEnumerable<ForeignKeyDefinition> targetReferencingForeignKeysInfo =
+                            listOfForeignKeys.Where(fk =>
+                                fk.ReferencingColumns.Count > 0
+                                && fk.ReferencedColumns.Count > 0
+                                && fk.Pair.ReferencingDbTable.Equals(targetDbo));
+
+                        ForeignKeyDefinition? sourceReferencingFKInfo = sourceReferencingForeignKeysInfo.FirstOrDefault();
```

Comment:
The lists `sourceReferencingForeignKeysInfo` and `targetReferencingForeignKeysInfo` can contain more than one element in the following scenario.

Consider a composite relationship field setup which is configured through 2 FK constraints. 
Ex: Table1 (A,B,C); Table2 (X,Y,Z) 
FK constraint 1: Table1.A --> Table2.X (Column A references column X)
FK constraint 2: Table2.B --> Table2.Y (Column B references column Y)

In this scenario, the list `sourceReferencingForeignKeysInfo` will contain 2 elements right?

Reply:
No, they will all be combined into one FK info.